contracts/staking/ServiceStakingBase.sol

```diff
@@ -393,11 +424,12 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
                 // Record the reward for the service if it has provided enough transactions
                 if (ratioPass) {
                     // Calculate the reward up until now and record its value for the corresponding service
-                    uint256 reward = rewardsPerSecond * ts;
-                    totalRewards += reward;
-                    eligibleServiceRewards[numServices] = reward;
+                    eligibleServiceRewards[numServices] = rewardsPerSecond * ts;
+                    totalRewards += eligibleServiceRewards[numServices];
```

Comment:
Wasn't the old implementation more gas efficient?

Reply:
Yes, we are just addressing comments for calculating reward for the current period, as Mariapia stated we wanted to have more information about rewards. I have a comment just above. Also, it's a view function that is not used on-chain by the staking contract.